capture/genBitmaps.js

```diff
@@ -169,10 +169,32 @@ function processScenario (casper, scenario, scenarioOrVariantLabel, scenarioLabe
       if (!scenario.hasOwnProperty('selectors')) {
         scenario.selectors = ['body'];
       }
+
+    // if similar selectors exist on the given page, expand and test them too
+    if (scenario.enableSelectorExpansion) {
+        scenario.selectors = scenario.selectors.reduce(function(acc, selector, index) { 
+            var expandedSelectors = casper.evaluate(function(selector, index) {
+                return [].slice.call(document.querySelectorAll(selector)).map(function(element, i) {
+
+                        // by using __n{index}__, we are making sure that this classname does not clash with user defined ones 
+                        var nPart = '__n' + index + '__' + i;
+                        element.classList.add(nPart);
+                        return selector + '.' + nPart;
+                });
+            }, selector, index);
+            return acc.concat(expandedSelectors);
+        }, []);
+    }
+
+
+
       scenario.selectors.forEach(function (o, i, a) {
         var cleanedSelectorName = o.replace(/[^a-z0-9_\-]/gi, ''); // remove anything that's not a letter or a number
         var switchedScenarioOrVariantLabel = (isReference) ? scenarioLabel : scenarioOrVariantLabel;
 
+        if (/__n\d*__\d*/.test(cleanedSelectorName)) {
```

Comment:
Instead of modifying `i` - why not  add a new template key `expandedIndex` and add that to the default template?  We should also make sure this works with the new and currently  undocumented `-- filter=scenarioId` feature.   Will using a filter put our expanded elements out of sync with their index?